### DISCUSSION TITLE
benchmark: 2 x Xeon E5-2660v3 - 20 cores, Ubuntu 22.04.3 LTS

### DISCUSSION THREAD

**Discussion by RedAndr** [leejet/stable-diffusion.cpp#70] [Benchmark]
SD 1.5 512x512
| Precision | GCC | ICC | ICC/BLAS |
|--------|--------|--------|--------|
| F32 | 252.34 | 243.62 | 342.08 |
| F16| 252.16 | 243.45 | 352.99 |
| Q8_0| 239.12 | 236.24 | 354.91 |
| Q4_1 | 246.04 | 237.86 | 353.44 |

Q8 is the fastest, but not significantly. Overall, there is barely any difference between precisions, which is kind of a surprize. To me it looks like code is always using F32. Used memory is about the same too: 2815, 2325, 2084, 1971 MB. Intel compiler is a little bit faster than GCC. OpenBLAS slows down a lot, wouldn't recommend it at all.

**Comment by JohnClaw** [comment]
> Intel compiler is a little bit faster than GCC.

Could you make and upload to github SD.made_with_Intel_compiler_AVX2.exe, please?

**Reply by RedAndr** [reply]
It's a Linux binary, not a Windows executable. Is it Ok?

**Comment by JohnClaw** [comment]
> It's a Linux binary, not a Windows executable. Is it Ok?

Not my cup of tea. I don't use Linux.

**Comment by Cyberhan123** [comment]
Is here benchmark with gguf? gguf with q8_0

**Reply by mchome** [reply]
Quantization doesn't make it faster unless the device get instructions.

**Reply by RedAndr** [reply]
It does for llama.cpp and whisper.cpp. For example, 8-bit quantization is twice as fast as the original 16-bit model. And 4-bit - almost 4x times. Of course, SD is much more sensitive to quantization, which degrades the image quality a lot, but at least it must be faster. Otherwise, there is no sense at all to use quantization. It's worse and not faster.
